File: src/ru/hudspeed/pro/ThemeMode.java
package ru.gpsantiradar.app;

public enum ThemeMode {
    LIGHT("Светлая"),
    DARK("Тёмная"),
    AUTOMATIC("Автоматически");

    private final String title;

    ThemeMode(String title) {
        this.title = title;
    }

    public String title() {
        return title;
    }

    public static ThemeMode fromStored(String value) {
        if (value != null) {
            for (ThemeMode mode : values()) {
                if (mode.name().equalsIgnoreCase(value)) return mode;
            }
        }
        return AUTOMATIC;
    }
}

File: src/main/kotlin/ru/gpsantiradar/app/ThemeMode.kt
package ru.gpsantiradar.app

enum class ThemeMode(private val title: String) {
    LIGHT("Светлая"), DARK("Тёмная"), AUTOMATIC("Автоматически");
    fun title(): String = title

    companion object {
        fun fromStored(value: String?): ThemeMode =
            entries.firstOrNull { it.name.equals(value, ignoreCase = true) } ?: AUTOMATIC
    }
}
